src/ai/backend/storage/api/storages.py

```diff
@@ -0,0 +1,398 @@
+"""
+S3 Storage API endpoints for file upload/download operations
+"""
+
+from __future__ import annotations
+
+import logging
+from http import HTTPStatus
+from typing import TYPE_CHECKING
+
+import trafaret as t
+from aiohttp import web
+from aiohttp.web_request import FileField
+from pydantic import ValidationError
+
+from ai.backend.common import validators as tx
+from ai.backend.common.dto.storage.request import S3TokenData
+from ai.backend.common.dto.storage.response import (
+    DeleteResponse,
+    ErrorResponse,
+    ObjectInfoResponse,
+    PresignedDownloadResponse,
+    PresignedUploadResponse,
+    UploadResponse,
+)
+from ai.backend.common.json import dump_json_str
+from ai.backend.logging import BraceStyleAdapter
+
+from ..client.s3 import S3Client
+from ..exception import (
+    PresignedDownloadURLGenerationError,
+    PresignedUploadURLGenerationError,
+    StorageObjectNotFoundError,
+    StorageProxyError,
+)
+from ..utils import log_client_api_entry
+
+if TYPE_CHECKING:
+    from ..context import RootContext
+
+log = BraceStyleAdapter(logging.getLogger(__spec__.name))
+
+_DEFAULT_CHUNKS = 8192  # Default chunk size for streaming uploads
+_DEFAULT_TOKEN_DURATION = 1800  # Default token expiration time in seconds
+
+s3_token_data_iv = t.Dict(
+    {
+        t.Key("op"): t.Enum(
+            "upload", "download", "info", "delete", "presigned_upload", "presigned_download"
+        ),
+        t.Key("bucket"): t.String(),
+        t.Key("key"): t.String(),
+        t.Key("expiration", optional=True): t.Int(gt=0, lte=604800),
+        t.Key("content_type", optional=True): t.String(),
+        t.Key("min_size", optional=True): t.Int(gte=0),
+        t.Key("max_size", optional=True): t.Int(gt=0),
+        t.Key("filename", optional=True): t.String(),
+    },
+).allow_extra("*")  # allow JWT-intrinsic keys
```

Comment:
With pydantic, isn't there no need to specifically use s3 token data iv?
How about reusing the `@pydantic_api_handler` that I worked on before?